Creator Features 2023-03-01-preview

ismael-ms opened the pull request:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [x] Other, please clarify - This is technically a new service onboarding. However, I wanted to clarify that this service is the successor of an existing service, [WFS](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/maps/data-plane/Creator/preview/2.0/wfs.json). We opted to create a whole new service instead of version as we wanted to rename the service to Features (as WFS is now a legacy service per Open Geospatial Consortium standards), as well as make a number of small changes to existing APIs that would sum up to breaking changes. Features keeps the same functionality as the read APIs in WFS while also adding a set of new APIs for write operations.
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
      - February 2023
  3. When do you expect to publish the swagger?
      - February 2023
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [x] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following appy to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

fix https://github.com/Azure/azure-rest-api-specs/issues/22058

ismael-ms committed:
Adds base for updating Creator from version preview/2.0 to version 2022-01-01-preview

ismael-ms committed:
Updates readme

ismael-ms committed:
Updates API version in new specs and examples

ismael-ms committed:
Remove all files in Creator/preview/2022-01-01-preview except for WFS

ismael-ms committed:
Created swagger docs for new WFS version and renamed WFS to Features

ismael-ms committed:
Update api-version in Features examples

ismael-ms committed:
Add link to Facility ontology for Features docs

ismael-ms committed:
Fix ModelValidation errors in Features

ismael-ms commented on specification/maps/data-plane/Creator/readme.md:
````diff
@@ -27,14 +27,27 @@ These are the global settings for Creator Client.
 ```
 
 
+### Tag: package-preview-2022-01
+
+These settings apply only when `--tag=package-preview-2022-01` is specified on the command line.
+
+```yaml $(tag) == 'package-preview-2022-01'
+input-file:
+  - preview/2022-01-01-preview/alias.json
+  - preview/2022-01-01-preview/data.json
+  - preview/2022-01-01-preview/dataset.json
+  - preview/2022-01-01-preview/dwgconversion.json
+  - preview/2022-01-01-preview/featurestate.json
+  - preview/2022-01-01-preview/tileset.json
+  - preview/2022-01-01-preview/wfs.json
+```
````
This is likely to change with the other API doc work going on in parallel. Please disregard the failed validation checks in the pipeline for the time being.

ismael-ms commented on specification/maps/data-plane/Creator/preview/2022-01-01-preview/features.json:
```diff
@@ -0,0 +1,959 @@
+            "$ref": "#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "../2.0/dataset.json#/parameters/DatasetId"
```
References to dataset.json will be updated to latest version pending doc changes.

ismael-ms committed:
Fix Creator readme.md

ismael-ms committed:
Fix Swagger LintDiff errors in Features

ismael-ms committed:
Note that featureId is case sensitive for Features

subbarayudukamma commented on specification/maps/data-plane/Creator/preview/2022-01-01-preview/examples/features/Features_GetCollection.json:
```diff
@@ -0,0 +1,34 @@
+        "ontology": "Facility-2.0",
+        "links": [
+          {
+            "href": "https://us.atlas.microsoft.com/wfs/dataset/218fda98-e638-0edf-5ef7-28491ff3bed4/collections/unit/definition?api-version=2022-01-01-preview",
```
WFS? What is it? :)

ismael-ms replied:
Aghh looks like I missed some references in the examples. Thanks for catching it! Fixed ✔️

subbarayudukamma commented on specification/maps/data-plane/Creator/preview/2022-01-01-preview/features.json:
```diff
@@ -0,0 +1,964 @@
+        }
+      },
+      "post": {
+        "description": "**Applies to:** see pricing [tiers](https://aka.ms/AzureMapsPricingTier).\n\nCreator makes it possible to develop applications based on your private indoor map data using Azure Maps API and SDK. [This](https://docs.microsoft.com/azure/azure-maps/creator-indoor-maps) article introduces concepts and tools that apply to Azure Maps Creator. For more information with regards to ...
```
Should we take another look at this boilerplate description? Is it really adding anything?

ismael-ms replied:
I'm personally a fan of the links in the description since they provide useful references if it's your first time reading about the service. I didn't want to rock the boat too much here, but I'm definitely open to suggestions!

brendansco replied:
Having the links here is useful in the scenario that someone searches for one of the APIs and wants to get to learn more about the Creator service.  However, I think that this is something that should be updated a bit, also maybe changing the order of things so that the overview page has content that is more useful.

ismael-ms replied:
Great suggestion on updating the order so that the overview page is more useful. I've now cleaned up and updated the description for all the Features APIs. Let me know what your thoughts are.

subbarayudukamma commented on specification/maps/data-plane/Creator/preview/2022-01-01-preview/features.json:
```diff
@@ -0,0 +1,964 @@
+        }
+      }
+    },
+    "FeatureResult": {
```
add azuremapsintid and featureid ?

ismael-ms replied:
Good question! If I'm not mistaken, azuremapsintid and featureid fall under the `properties` field of a feature which is already defined as part of `/common.json#/definitions/GeoJsonFeature`, so I don't think we need to specifically single out those properties.

ismael-ms committed:
Change 'wfs' mentions to 'features'

ismael-ms committed:
Reformatted description for Features APIs

ismael-ms committed:
Remove featureType attribute from Features

ismael-ms committed:
Merge branch 'main' into Creator-2022-01-01-preview-features

ismael-ms committed:
Move Features spec into version preview 2022-09-01

ismael-ms committed:
Update api version to 2022-09-01

ismael-ms committed:
Move features directory into examples

ismael-ms committed:
Update dataset references to latest version

ismael-ms committed:
Updated the Features examples to conformed schema

ismael-ms committed:
Updated collections response schema to conform to ogc

ismael-ms committed:
Remove hreflang and type fields from ref links

ismael-ms committed:
Fix linting issues

ismael-ms committed:
Add format to integer types

ismael-ms committed:
Fix noun_verb convention

ismael-ms committed:
Add missing operationId for landing page

ismael-ms committed:
Adding Get API Definition

ismael-ms committed:
Fix linting issues from GetAPIDefinition

ismael-ms committed:
Resolve model validation errors with GetAPIDefinition

ismael-ms committed:
Rename APIDefinition to ApiDefinition to fix title formatting

tejitpabari99 commented on specification/maps/data-plane/Creator/preview/2022-09-01-preview/features.json:
```diff
@@ -0,0 +1,959 @@
+        }
+      },
+      "post": {
+        "description": "The Post Feature API creates a feature with the provided attributes in the given collection.\n\nThe Features API is a part of [Creator](https://docs.microsoft.com/azure/azure-maps/creator-indoor-maps), which makes it possible to develop applications based on your private indoor map data using Azure Maps API and SDK. Features API follows the [Open Geospatial Consortium API  ...
```
Request Body for Features Create should have a row for Properties. 
Similarly, Features definition should have a row for Properties.

ismael-ms replied:
I'm a bit confused here, which lines are you referring to exactly?

tejitpabari99 replied:
I was referring to this section:
![image](https://user-images.githubusercontent.com/35284488/210878257-9d138147-b7ef-4de4-8c86-d6436afe9ce2.png)
Not sure where it is located. 
For Features, I was referring to this section:
![image](https://user-images.githubusercontent.com/35284488/210878460-cf86a3b5-5332-40ff-8f6f-e05cb526d57d.png)

Same applies to other comments

ismael-ms replied:
So after doing some digging, it appears that the request body definition I defined in [#parameters/FeatureBody](https://github.com/Azure/azure-rest-api-specs/pull/19539/files#diff-bebed0270ce3522f925f7208ea9dce74c2e5d8faaecc10fc0a59c12a9e9dc9b9R116) isn't being interpreted correctly. I'm still investigating what the error is, so pending updates inbound.

ismael-ms replied:
Found the issue and now properties is rendered properly. Thanks for catching that!

tejitpabari99 commented on specification/maps/data-plane/Creator/preview/2022-09-01-preview/features.json:
```diff
@@ -0,0 +1,959 @@
+        }
+      },
+      "post": {
+        "description": "The Post Feature API creates a feature with the provided attributes in the given collection.\n\nThe Features API is a part of [Creator](https://docs.microsoft.com/azure/azure-maps/creator-indoor-maps), which makes it possible to develop applications based on your private indoor map data using Azure Maps API and SDK. Features API follows the [Open Geospatial Consortium API  ...
```
Is it necessary to define the different geometry types, after we have linked to the Geojson Doc. I.e., can we remove this: 
> GeoJsonGeometry:
GeoJsonPoint
GeoJsonMultiPoint
GeoJsonLineString
GeoJsonMultiLineString
GeoJsonPolygon
GeoJsonMultiPolygon
GeoJsonGeometryCollection

because we have a link to [RFC 7946](https://tools.ietf.org/html/rfc7946#section-3.1) in the description (under Request Body and under Feature Definition)

ismael-ms replied:
These definitions are coming from [common.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/maps/data-plane/Common/preview/1.0/common.json#L184). Although it may be overly precise, since it's a commonly referenced file, I think it's best to keep it consistent across our different swagger docs.

tejitpabari99 commented on specification/maps/data-plane/Creator/preview/2022-09-01-preview/features.json:
```diff
@@ -0,0 +1,959 @@
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/FeatureResult"
```
Should we document what exact links should be in Feature Result. 
Eg: GET Collections will always return the specific 3 links. Similarly, GET Feature would also return specific links. Maybe documenting them would help users determine how to structure their code, rather than them making calls and figuring out what links are returned when and what to do with them.

ismael-ms replied:
As discussed offline, I've updated the description of the FeaturesEndpointLink definition to reference the OGC spec where there's more details about the link relations.

ismael-ms commented on specification/maps/data-plane/Creator/preview/2022-09-01-preview/features.json:
```diff
@@ -0,0 +1,959 @@
+        }
+      },
+      "post": {
+        "description": "The Post Feature API creates a feature with the provided attributes in the given collection.\n\nThe Features API is a part of [Creator](https://docs.microsoft.com/azure/azure-maps/creator-indoor-maps), which makes it possible to develop applications based on your private indoor map data using Azure Maps API and SDK. Features API follows the [Open Geospatial Consortium API  ...
```
Confused here as well, what lines are you referring to exactly?

tejitpabari99 replied:
Replied in the previous comment

ismael-ms replied:
Fixed the error here as well

ismael-ms commented on specification/maps/data-plane/Creator/preview/2022-09-01-preview/features.json:
```diff
@@ -0,0 +1,959 @@
+  "paths": {
+    "/features/datasets/{datasetId}": {
+      "get": {
+        "description": "The Get Landing Page API provides links to the API definition, the Conformance statements, and the metadata about the feature data in this dataset.\n\nThe Features API is a part of [Creator](https://docs.microsoft.com/azure/azure-maps/creator-indoor-maps), which makes it possible to develop applications based on your private indoor map data using Azure Maps API and SDK. Fe ...
```
I don't believe that page can be customized, at least I haven't seen any of our docs doing that. We had a previous discussion about the redundancy before and decided to keep it, so that anyone that stumbles upon a specific page can have a reference to others. See https://github.com/Azure/azure-rest-api-specs/pull/19539#discussion_r903177508

ismael-ms committed:
Updated the general description of the APIs

ismael-ms committed:
Update common references to version 2022-09-01-preview

ismael-ms commented on specification/maps/data-plane/Creator/preview/2022-09-01-preview/features.json:
```diff
@@ -0,0 +1,959 @@
+  "paths": {
+    "/features/datasets/{datasetId}": {
+      "get": {
+        "description": "The Get Landing Page API provides links to the API definition, the Conformance statements, and the metadata about the feature data in this dataset.\n\nThe Features API is a part of [Creator](https://docs.microsoft.com/azure/azure-maps/creator-indoor-maps), which makes it possible to develop applications based on your private indoor map data using Azure Maps API and SDK. Fe ...
```
Good suggestion, updated ✔️

ismael-ms commented on specification/maps/data-plane/Creator/preview/2022-09-01-preview/features.json:
```diff
@@ -0,0 +1,959 @@
+  "paths": {
+    "/features/datasets/{datasetId}": {
+      "get": {
+        "description": "The Get Landing Page API provides links to the API definition, the Conformance statements, and the metadata about the feature data in this dataset.\n\nThe Features API is a part of [Creator](https://docs.microsoft.com/azure/azure-maps/creator-indoor-maps), which makes it possible to develop applications based on your private indoor map data using Azure Maps API and SDK. Fe ...
```
Updated with some links and further clarification on meanings.

ismael-ms commented on specification/maps/data-plane/Creator/preview/2022-09-01-preview/features.json:
```diff
@@ -0,0 +1,959 @@
+  "paths": {
+    "/features/datasets/{datasetId}": {
+      "get": {
+        "description": "The Get Landing Page API provides links to the API definition, the Conformance statements, and the metadata about the feature data in this dataset.\n\nThe Features API is a part of [Creator](https://docs.microsoft.com/azure/azure-maps/creator-indoor-maps), which makes it possible to develop applications based on your private indoor map data using Azure Maps API and SDK. Fe ...
```
Updated as well, thanks for the suggestion.

subbarayudukamma reviewed (commented):
Will add more as I spend more time.

ismael-ms committed:
Update the description of Patch and Put

ismael-ms committed:
Fix POST, PUT, PATCH request bodies

ismael-ms commented on specification/maps/data-plane/Creator/preview/2022-09-01-preview/features.json:
```diff
@@ -0,0 +1,1067 @@
+        }
+      },
+      "put": {
+        "description": "The Put Feature API updates the feature identified by the provided id in the given collection.\n\nThe Features API is a part of [Creator](https://docs.microsoft.com/azure/azure-maps/creator-indoor-maps), which makes it possible to develop applications based on your private indoor map data using Azure Maps API and SDK. Features API follows the [Open Geospatial Consortium AP ...
```
Great point! I've updated the description for both to be clearer on functionality. Please let me know your thoughts.

ismael-ms commented on specification/maps/data-plane/Creator/preview/2022-09-01-preview/features.json:
```diff
@@ -0,0 +1,1067 @@
+        }
+      },
+      "post": {
+        "description": "The Post Feature API creates a feature with the provided attributes in the given collection.\n\nThe Features API is a part of [Creator](https://docs.microsoft.com/azure/azure-maps/creator-indoor-maps), which makes it possible to develop applications based on your private indoor map data using Azure Maps API and SDK. Features API follows the [Open Geospatial Consortium API  ...
```
Hmm I'm a bit conflicted here because GET, PUT, and PATCH API descriptions all start with the http verb, so using POST here remains consistent across. What are your thoughts on how to keep it consistent?

subbarayudukamma replied:
I don't think the consistency was intentional. If someone is reading just this description I don't think it will make sense. I think it can be either "This Post Feature API creates " or "The Create Feature API creates...".

ismael-ms replied:
I've updated it to "This Post Feature API creates..."

ismael-ms committed:
Minor updates to descriptions

heaths commented on specification/maps/data-plane/Creator/readme.md:
```diff
@@ -44,6 +44,7 @@ input-file:
   - preview/2022-09-01-preview/style.json
   - preview/2022-09-01-preview/mapconfiguration.json
   - preview/2022-09-01-preview/dataset.json
+  - preview/2022-09-01-preview/features.json
```
Add new APIs to an existing version - preview or stable - violates our [Breaking Changes Policy](https://aka.ms/AzBreakingChangesPolicy). Feel free to come to our office hours to discuss this further.

ismael-ms replied:
Thanks for the feedback. We will be attending Monday's office hours to discuss this portion.

ismael-ms replied:
Hey @heaths, per discussions with issue https://github.com/Azure/azure-rest-api-specs/issues/22058 and PR https://github.com/Azure/azure-rest-api-specs/pull/22590, I've merged this branch with the 2023-03-01-preview feature branch. This now has all the latest changes and is ready for review again under the guise that it will be released as part of 2023-03-01-preview.

heaths commented on specification/maps/data-plane/Creator/preview/2022-09-01-preview/features.json:
```diff
@@ -0,0 +1,1074 @@
+          "description": "The OpenAPI components object.",
+          "type": "object",
+          "properties": {
+            "parameters": {
+              "description": "The OpenAPI parameters object.",
+              "type": "object"
```
Should this have `"additionalProperties": true` to denote it's a map (JSON object) and accepts more information? This will also help our code generation for SDKs.

ismael-ms replied:
Great suggestion, thank you! I've updated it.

heaths commented on specification/maps/data-plane/Creator/preview/2022-09-01-preview/features.json:
```diff
@@ -0,0 +1,1074 @@
+            }
+          }
+        },
+        "tags": {
+          "description": "The OpenAPI tags object.",
+          "type": "array",
+          "items": {
+            "type": "object",
+            "properties": {
+              "name": {
+                "description": "The name of the tag.",
+                "type": "string",
+                "example": "Features"
+              },
+              "description": {
+                "description": "The description of the tag.",
+                "type": "string",
+                "example": "Access to data (features)."
+              }
+            }
+          }
+        }
```
Most times tags are just an object with `additionalProperties: true` e.g.,

```json
{
  "tags": {
    "foo": "1",
    "bar": "b"
  }
}
```

Unless this is conforming to a standard? Also, where is the value? You only document a name and description, or is this more like metadata and instances will have a version?

Also seems like `name` should be required.

ismael-ms replied:
So this is the response definition for the GET /api, which is a bit interesting to document as it's returning the API definition (swagger) of the Features service. This API is required by the OGC standard as described here: https://docs.ogc.org/is/17-069r4/17-069r4.html#_api_definition_2. The tags are a list of metadata, and thanks for catching the name key being required. I've updated that portion.

heaths replied:
Thanks for clarifying.

heaths commented on specification/maps/data-plane/Creator/preview/2022-09-01-preview/features.json:
```diff
@@ -0,0 +1,1074 @@
+        "href"
+      ],
+      "properties": {
+        "href": {
```
This probably should be annotated with `"format": "uri"`.

ismael-ms replied:
Good catch, updated ✔️

ismael-ms committed:
Address PR feedback: use additionalProperties and format uri

ruowan commented:
/azp run

ismael-ms committed:
Update the API descriptions to reference the appropriate OGC specifications

ismael-ms committed:
Update the description of the Delete API

heaths commented:
@ismael-ms if possible, can you please avoid rebasing and force-pushing. It makes it very difficult to simply view the changes since last we reviewed, thus having to review the entire PR again and taking longer.

ismael-ms commented:
> @ismael-ms if possible, can you please avoid rebasing and force-pushing. It makes it very difficult to simply view the changes since last we reviewed, thus having to review the entire PR again and taking longer.
Understood. Just wanted to fix a mistake in the previous commit, but I'll refrain from cleaning up history. Apologies for the trouble.

ismael-ms committed:
Merge branch 'release-maps-Creator-2023-03-01-preview-feature-branch' into Creator-2022-01-01-preview-features

ismael-ms committed:
Move Features docs to version 2023-03-01-preview

ismael-ms committed:
Update version parameters and remove 'id' from Features POST and PUT request body

ismael-ms committed:
Remove default api-version in query parameters

ismael-ms commented:
@mikekistler, @JeffreyRichter, @heaths Thank you all for your assistance and guidance in developing a new API version. I have addressed the versioning concerns in this PR now and it is ready for review again.

For reference, here's a quick summary of things that have been changed since the last review:
- I merged [release-maps-Creator-2023-03-01-preview-feature-branch](https://github.com/Azure/azure-rest-api-specs/tree/release-maps-Creator-2023-03-01-preview-feature-branch) into this branch to have the latest 2023-03-01-preview changes. (https://github.com/Azure/azure-rest-api-specs/pull/19539/commits/66ffa06dd5040e4907d438d4e4229c076d2b4f9a)
- I updated the API version for Features to 2023-03-01-preview (https://github.com/Azure/azure-rest-api-specs/pull/19539/commits/41693437b3ef5e1e64a2d4e2550f1795bfcc1609 and https://github.com/Azure/azure-rest-api-specs/pull/19539/commits/ec3fb5434a4f8e0c73e34aa3c80cfd6e2c2dbb89)
- I removed the "id" field in the POST (create feature) and PUT (update feature) request body. This is still conforming with the Open Geospatial Consortium specification, and it is not a breaking change since it is a new API. (https://github.com/Azure/azure-rest-api-specs/pull/19539/commits/ec3fb5434a4f8e0c73e34aa3c80cfd6e2c2dbb89)
- Removed the default api-version since it is a required field. (https://github.com/Azure/azure-rest-api-specs/pull/19539/commits/284d62c7a6e19ca10ed41e9c8ae4327bdf78cc58)

Please let me know if there are any questions. I'll be happy to answer 😃

heaths commented on specification/maps/data-plane/Creator/preview/2023-03-01-preview/features.json:
```diff
@@ -120,7 +119,7 @@
       "x-ms-parameter-location": "client",
       "schema": {
-        "$ref": "#/definitions/Feature"
+        "$ref": "#/definitions/FeatureRequestBody"
```
Nit: I'd drop "Body". You wouldn't want to see this in an SDK model, right? `FeatureRequest` seems reasonable.

ismael-ms replied:
Thanks for the feedback Heath! Since we're shipping soon, we can proceed with it as is. Please assist us in merging when you have the opportunity.

ismael-ms replied:
Since I was already making changes for the naming of patch and put APIs, I went back and fixed this as well. Thanks again!

heaths reviewed (approved):
One suggestion, but otherwise LGTM. Let me know if you plan to change that; otherwise, I can merge it.

mikekistler commented:
@jhendrixMSFT This PR is ready to merge.

jhendrixMSFT commented on specification/maps/data-plane/Creator/preview/2023-03-01-preview/features.json:
```diff
@@ -0,0 +1,1089 @@
+      },
+      "put": {
+        "description": "This Put Feature API conforms with the **drafted** [Open Geospatial Consortium API - Features - Part 4: Create, Replace, Update and Delete](http://docs.opengeospatial.org/DRAFTS/20-002.html) document. This API **replaces** the feature identified by the provided id in the given collection, i.e., the request body overwrites all the existing attributes of a feature.\n\nThe Fe ...
+        "operationId": "Features_Put",
```
We typically refrain from using the verb as the name of the API.  What about Replace?  Seems to match with the description.

ismael-ms replied:
Thanks for the feedback! It is now fixed.

jhendrixMSFT commented on specification/maps/data-plane/Creator/preview/2023-03-01-preview/features.json:
```diff
@@ -0,0 +1,1089 @@
+      },
+      "patch": {
+        "description": "This Patch Feature API conforms with the **drafted** [Open Geospatial Consortium API - Features - Part 4: Create, Replace, Update and Delete](http://docs.opengeospatial.org/DRAFTS/20-002.html) document. This API **updates** attributes by performing a merge patch on the feature identified by the provided id in the given collection. It allows for the update of individual fea ...
+        "operationId": "Features_Patch",
```
We typically refrain from using the verb as the name of the API.  What about Update?  Seems to match with the description.

ismael-ms replied:
Thanks again, I've fixed it here as well.

ismael-ms committed:
Rename FeatureRequestBody definition to FeatureRequest

ismael-ms committed:
Rename patch and put API to update and replace
